addons/web_editor/models/ir_qweb_fields.py

```diff
@@ -155,10 +155,13 @@ def attributes(self, record, field_name, options, values):
             attrs['placeholder'] = placeholder
 
         if options['translate'] and field.type in ('char', 'text'):
-            name = "%s,%s" % (record._name, field_name)
-            domain = [('name', '=', name), ('res_id', '=', record.id), ('type', '=', 'model'), ('lang', '=', options.get('lang'))]
-            translation = record.env['ir.translation'].search(domain, limit=1)
-            attrs['data-oe-translation-state'] = translation and translation.state or 'to_translate'
+            lang = record.env.lang or 'en_US'
+            if lang == 'en_US':
+                attrs['data-oe-translation-state'] = 'translated'
+            else:
+                value_en = record.with_context(lang='en_US')[field_name]
+                value_lang = record.with_context(lang=lang)[field_name]
+                attrs['data-oe-translation-state'] = 'translated' if value_en != value_lang else 'to_translate'
```

Comment:
The root problem is: we use 'en_US' as both English and fallback language
when we use jsonb to store translations {'en_US': 'value1', 'fr_FR': 'value2'}
We don't store 'to_translate'/'translated'
So, we don't know if the value1 is really English or not

I agree with you it is disapointing that the green/yellow hint is not perfect
and maybe we need a new UI for the hint in the future
But for the current stage, what I can do best is to comparing the value with 'en_US'
to guess if it is translated
Or a simple improvement is:
I compare the value with the main lang value for website

If it works for you, could you tell me how to get the main lang for website?

Reply:
I propose to just live with it, for now on. It's not perfect, but it's not a big deal neither. (I can clearly see in a page what's translated or not, by reading the text)

Reply:
> I can clearly see in a page what's translated or not, by reading the text

That's not entirely true, it at least can't always be "clearly seen".
Try find the only 2 untranslated sentence in a whole blog post where most of the blog post is translated, you will be having a hard time and losing time for no valid reason without the visual hint (and worst, maybe even missing some).

Maybe when doing a minimal demo with 2 text block it is easy, but not in every cases. That's why we have that visual helper.

> That only happens if your website is managed in a non english lang, and you translate english.

"**Only**"..
Isn't that like a very common case for websites over the world created outside of UK and US?
Creating a website and/or an ecommerce in your country lang and then translating it in English for everyone to have a lang to fallback on is not exactly rare.

---------

But we plan to change that behavior anyway[*] so you can ignore it like @fpodoo said since it is not a big deal.
We will have a deeper look at the whole translation feature/behavior later.

[*] There is other issues about it, mainly some conflicts with the editor where the background color is kept on save.

Reply:
Just a small opinion, there are issues with translate mode that we have to change/fix anyway but:

> That only happens if your website is managed in a non english lang, and you translate english.

= All international companies that start using Odoo from a non-English language country, right? :grimacing: Or simply companies that might want non native-speakers to be able to visit their website and use English. Ok, English is probably the most used language as default for a website in Odoo (is it even reaching 50% though?) but the very large majority of the rest falls in the case "That only happens if your website is managed in a non english lang, and you translate english." I think? Maybe there are even more of those than english as default language TBH, I would not be surprised.